websocket-ballerina/Package.md

```diff
@@ -1,32 +1,33 @@
 ## Module Overview
 
-This module provides an implementation for connecting and interacting with WebSocket endpoints. The module facilitates two types of network entry points as ‘Client’ and ‘Listener’.
+This module provides an implementation for connecting and interacting with WebSocket endpoints. The module facilitates three types of network entry points as ‘AsyncClient’, ‘SyncClient’ and ‘Listener’.
 
-### WebSocket Module
+The `AsyncClient` has a callback service that can be registered at the initialization of the client. It has a fixed set of remote methods in this service and they get called on the receipt of messages from the server. The ‘SyncClient’ also can have a callback service registered to receive control frames like ping, pong and close.
 
-This module also provides support for WebSockets. There are two types of WebSocket endpoints: `WebSocketClient` and `WebSocketListener`. Both endpoints support all WebSocket frames. The `WebSocketClient` has a callback service.
+In the server side, initial websocket service is there to handle upgrade requests. It has a single `onUpgrade` resource which takes in `http:Caller` and `http:Request` optionally. The `onUpgrade` resource returns a `websocket:Service` to which incoming messages gets dispatched after a successful websocket connection upgrade. This resource can be used to intercept the initial HTTP upgrade with custom headers or to cancel the websocket upgrade by returning an error.
+The returning `websocket:Service` has a fixed set of remote methods.
 
-There are two types of services for WebSockets. The service of the server has the `WebSocketCaller` as the resource parameter and the callback service of the client has `WebSocketClient` as the resource parameter. The WebSocket services have a fixed set of resources that do not have a resource config. The incoming messages are passed to these resources.
-
-**WebSocket upgrade**: During a WebSocket upgrade, the initial message received is an HTTP request. To intercept this request and perform the upgrade explicitly with custom headers, the user must create an HTTP resource with WebSocket-specific configurations as follows:
+**WebSocket upgrade**: During a WebSocket upgrade, the initial message received is an HTTP request. 
 
 ```ballerina
-@http:ResourceConfig {
-    webSocketUpgrade: {
-        upgradePath: "/{name}",
-        upgradeService: chatApp
-    }
-}
-resource function upgrader(http:Caller caller, http:Request req, string name) {
+service /basePath on new websocket:Listener(21003) {
```

Comment:
If the base path is not given does it default to /ws ?

Reply:
The default is '/'. `ws://localhost:21029` or `wss://localhost:21029/` will get dispatched to this service.

Reply:
I think it should default to /ws as that is most common case ? wdyt ?

Reply:
As per the discussion had, we decided not to do this for now.